src/main/kotlin/org/rust/ide/navigation/goto/RsCurrentActionService.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.navigation.goto
+
+import com.intellij.codeInsight.navigation.actions.GotoDeclarationAction
+import com.intellij.codeInsight.navigation.actions.GotoDeclarationOnlyAction
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.actionSystem.ex.AnActionListener
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+
+@Service
+class RsCurrentActionService {
+    var isGoToDeclarationAction: Boolean = false
+        private set
+
+    private class Listener : AnActionListener {
+        override fun beforeActionPerformed(action: AnAction, dataContext: DataContext, event: AnActionEvent) {
+            if (action is GotoDeclarationAction || action is GotoDeclarationOnlyAction) {
```

Comment:
Let's extract this condition into property/function not to duplicate logic

Something like

```kotlin
    private val AnAction.isGoToDeclarationAction: Boolean get() = 
        this is GotoDeclarationAction || this is GotoDeclarationOnlyAction
```